src/metabase/serialization/dump.clj

```diff
@@ -0,0 +1,294 @@
+          (:name table)))
+
+(defmethod fully-qualified-name (type Field)
+  [field]
+  (str (->> field :table_id Table fully-qualified-name) "/fields/" (:name field)))
+
+(defmethod fully-qualified-name (type Metric)
+  [metric]
+  (str (->> metric :table_id Table fully-qualified-name) "/metrics/" (:name metric)))
+
+(defmethod fully-qualified-name (type Segment)
+  [segment]
+  (str (->> segment :table_id Table fully-qualified-name) "/segments/" (:name segment)))
+
+(defmethod fully-qualified-name (type Collection)
+  [collection]
+  (let [parents (->> (str/split (:location collection) #"/")
+                     rest
+                     (map #(-> % Integer/parseInt Collection :name (str "/collections")))
+                     (str/join "/")
+                     (format "%s/"))]
+    (str "/collections/collections/" parents (:name collection))))
+
+(defmethod fully-qualified-name (type Dashboard)
+  [dashboard]
+  (format "%s/dashboards/%s"
+          (or (some->> dashboard :collection_id Collection fully-qualified-name)
+              "/collections")
+          (:name dashboard)))
+
+(defmethod fully-qualified-name (type Pulse)
+  [pulse]
+  (format "%s/pulses/%s"
+          (or (some->> pulse :collection_id Collection fully-qualified-name)
+              "/collections")
+          (:name pulse)))
+
+(defmethod fully-qualified-name (type Card)
+  [card]
+  (format "%s/cards/%s"
+          (or (some->> card
+                       :dataset_query
+                       qp.util/query->source-card-id
+                       Card
+                       fully-qualified-name)
+              (some->> card
+                       :collection_id
+                       Collection
+                       fully-qualified-name)
+              "/collections")
+          (:name card)))
+
+(defmethod fully-qualified-name nil
+  [_]
+  nil)
+
+(def ^:private SegmentOrMetric
+  [(s/one (s/constrained su/KeywordOrString
+                         (comp #{:metric :segment} qp.util/normalize-token))
+          "head")
+   (s/cond-pre s/Int su/KeywordOrString)])
+
+(def ^{:arglists '([form])} entity-reference?
+  "Is given form an MBQL entity reference?"
+  (some-fn field-reference? (complement (s/checker SegmentOrMetric))))
+
+(defn- entity-reference->fully-qualified-name
+  [[op & args :as entity-reference]]
+  (case (qp.util/normalize-token op)
+    :metric        [op (fully-qualified-name (Metric (first args)))]
+    :segment       [op (fully-qualified-name (Segment (first args)))]
+    :field-id      [op (if (string? (first args))
+                         (first args)
+                         (fully-qualified-name (Field (first args))))]
+    :fk->          (into [op]
+                         (for [arg args]
+                           (if (number? arg)
+                             (Field arg)
+                             (entity-reference->fully-qualified-name arg))))
+    :field-literal entity-reference))
+
+(defn- humanize-entity-references
```

Comment:
Why is this called "humanize"? What are we humanizing?

Reply:
Making IDs human readable (ie. path-like). Gladly change the name to something better, but I was drawing a blank.